libs/design-system/src/navigation/NavMenu.tsx

```diff
@@ -1,40 +1,106 @@
-import React, { ReactNode } from 'react';
+import React, { ReactNode, useState } from 'react';
 import { NavLink } from 'react-router-dom';
-import { Group } from '@mantine/core';
+import { Group, Transition, Popover } from '@mantine/core';
 import useStyles from './NavMenu.styles';
+import usePopoverStyles from './NavLinkPopover.styles';
 
 interface INavMenuProps {
   menuItems: {
     icon: ReactNode;
     label: string;
     link: string;
     testId?: string;
-    rightSide?: ReactNode;
+    rightSide?: ReactNode | { component: ReactNode; displayOnHover: boolean };
     condition?: boolean;
+    tooltipLabel?: string;
   }[];
 }
 
 export function NavMenu({ menuItems }: INavMenuProps) {
   const { classes } = useStyles();
+  const { classes: popoverClasses } = usePopoverStyles();
```

Comment:
this should be close to the extracted component, no need to propagate through the props

Reply:
Agree i have another issue that was solved but your other comment is by separating NavLink so we won't have infinite re-rendering.